src/company/company.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Inject, Injectable, NotFoundException } from "@nestjs/common";
+import { UpdateCompanyDto } from "./dtos/updateCompany.dto";
+import { GetCompanyQueryDto } from "./dtos/companyGetQuery.dto";
+import { FindOptions, Transaction } from "sequelize";
+import { CompanyModel, JobModel } from "../db/models";
+import { parseFilter, parseOrder, parsePagesize } from "../utils";
+import { COMPANY_DAO } from "../constants"; // Assuming you have these DTO
+
+@Injectable()
+export class CompanyService {
+  constructor(@Inject(COMPANY_DAO) private companyRepo: typeof CompanyModel) {}
+
+  async getAllCompanies(where: GetCompanyQueryDto) {
+    const findOptions: FindOptions<CompanyModel> = {
+      include: [
```

Comment:
This is a nested object of related columns and tables on which filtering could be done.

In this case, we don't need to filter companies based on the job because each company could have multiple jobs, and each job will have exactly one company, so filtering doesn't make sense here.